daras_ai_v2/gdrive_downloader.py

```diff
@@ -81,17 +94,60 @@ def gdrive_download(f: furl, mime_type: str) -> tuple[bytes, str]:
             fileId=file_id,
             mimeType=mime_type,
         )
+    return request, mime_type
+
+
+def download_blob_file_content(
+    service, request, file_id: str, f: furl, mime_type: str, export_links: dict
+) -> tuple[bytes, str]:
+    from googleapiclient.http import MediaIoBaseDownload
+    from googleapiclient.errors import HttpError
+
     # download
     file = io.BytesIO()
     downloader = MediaIoBaseDownload(file, request)
-    done = False
-    while done is False:
-        _, done = downloader.next_chunk()
-        # print(f"Download {int(status.progress() * 100)}%")
-    f_bytes = file.getvalue()
+
+    if (
+        mime_type
+        == "application/vnd.openxmlformats-officedocument.presentationml.presentation"
+    ):
+        # logger.debug(f"Downloading {str(f)!r} using export links")
+        f_url_export = export_links.get(mime_type, None)
+        if f_url_export:
+
+            f_bytes = download_from_exportlinks(f_url_export)
+        else:
+            request = service.files().get_media(
+                fileId=file_id,
+                supportsAllDrives=True,
+            )
+            downloader = MediaIoBaseDownload(file, request)
+
+            done = False
+            while done is False:
+                _, done = downloader.next_chunk()
+                # print(f"Download {int(status.progress() * 100)}%")
+            f_bytes = file.getvalue()
+
+    else:
+        done = False
+        while done is False:
+            _, done = downloader.next_chunk()
+            # print(f"Download {int(status.progress() * 100)}%")
+        f_bytes = file.getvalue()
+
     return f_bytes, mime_type
 
 
+def download_from_exportlinks(f: furl) -> bytes:
+    try:
+        r = requests.get(f)
+        f_bytes = r.content
```

Comment:
use raise_for_status() like we do everywhere else in code

Reply:
this likely doesnt support downloading private docs without auth, I may be wrong